R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,236 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Converting cotton back to primary equivalent (seed cotton)
+    # Seed cotton has no price in PRICESTAT. Need to derive its price from cotton lint and cottonseed
+    # Assigning a price for game meat so that OtherMeat is assigned a price
+    FAO_USA_ag_an_P_USDt_PRICESTAT %>%
+      select(-country.codes, -item.codes, -element, -element.codes) %>%
+      gather(year, price, -countries, -item) %>%
+      filter(item == "Seed cotton" | item == "Cotton lint" | item == "Cottonseed" | item == "Game meat" | item == "Cattle meat") %>%
+      # Modify item names to one word so that they can be used as column names when spreading item and doing calculations
+      mutate(item = sub(" ", "_", item)) %>%
+      spread(item, price) %>%
+      mutate(Seed_cotton = Cotton_lint * CONV_COTTON_LINT + Cottonseed * (1 - CONV_COTTON_LINT)) %>%
+      mutate(Game_meat = Cattle_meat) %>%
+      gather(item, price, -countries, -year) %>%
+      # Change item names back to original for mapping GCAM commodities
+      mutate(item = sub("_", " ", item)) ->
+      extra_price
+    # Put these calculated prices back to the dataset
+    FAO_USA_ag_an_P_USDt_PRICESTAT %>%
+      select(-country.codes, -item.codes, -element, -element.codes) %>%
+      gather(year, price, -countries, -item) %>%
+      filter(!(item == "Seed cotton" | item == "Cotton lint" | item == "Cottonseed" | item == "Game meat" | item == "Cattle meat")) %>%
+      bind_rows(extra_price) %>%
+      # Calculate unweighted averages for each FAO commodity over price years
+      filter(year %in% MODEL_PRICE_YEARS) %>%
+      group_by(countries, item) %>%
+      summarise_at(vars(price), mean, na.rm = TRUE) %>%
+      ungroup() %>%
+      filter(!is.na(price)) ->
+      Avg_price
+
+    # Computing average prices and production quantities of all commodities
+    # Subset only the relevant country/item combinations from the ag prodstat database
+    # NOTE: This mostly excludes fodder crops and "not elsewhere specified" crops. It also excludes about seven crops that are
+    # minor enough to not distort price estimates, but their omission may cause problems if a large number of commodities is used
```

Comment:
This is just pointing out that not all of the FAO crops listed in PRODSTAT (i.e., database of production and harvested area) are in FAO PRICESTAT (database of prices). Feel free to print out a list of them, but my guess is it's all either crop composites like "cereals, NES" or minor crops like Pomegranates that we would never have as GCAM commodities anyway.